lib/ruby_lsp/requests/document_symbol.rb

```diff
@@ -163,10 +182,15 @@ def on_module_node_enter(node)
       sig { params(node: Prism::DefNode).void }
       def on_def_node_enter(node)
         receiver = node.receiver
+        previous_symbol = @stack.last
 
         if receiver.is_a?(Prism::SelfNode)
           name = "self.#{node.name}"
-          kind = Constant::SymbolKind::METHOD
+          kind = Constant::SymbolKind::FUNCTION
+        elsif previous_symbol.is_a?(Interface::DocumentSymbol) && previous_symbol.name.start_with?("<<")
+          _operator, expression = previous_symbol.name.split(" ")
+          name = "(#{expression}.)#{node.name}"
```

Comment:
I am not sure adding the `(expression.)` prefix makes this more readable. Since these methods will be listed under the `<< expression` symbol anyway, I think it is fine to just list them with `node.name` only.

RubyMine seems to do it like that too, btw.

Reply:
@nirvdrum since you originally reported this, do you have any preferences / opinions? I'm fine either way.

Reply:
In the document symbol I'd just have `node.name` but I do see value in inline hits given some of those `class << self` can span multiple lines and you can't actually see anymore a method is inside a `class << self`. I do understand it is confusing though since `class << self; def self.foo; end` has a different result to `class << self; def foo; end`. Maybe there is other kind of hint that would make easy to know exactly that method is.